Tests/SerializationTransformerTests.swift

```diff
@@ -0,0 +1,215 @@
+                }
+            }
+            
+            context("Array") {
+                it("should transform the keys") {
+                    let object = [friend]
+                    let serialized = UppercaseTransformer(wrapped: object).serialize() as! [[String: AnyObject]]
+                    expect(serialized.first?["FRIENDS"]).toNot(beNil())
+                }
+            }
+            
+            context("Dictionary") {
+                it("should transform the keys") {
+                    let object = ["lowercase": friend]
+                    let serialized = UppercaseTransformer(wrapped: object).serialize() as! [String: [String: AnyObject]]
+                    expect(serialized["LOWERCASE"]?["FRIENDS"]).toNot(beNil())
+                }
+            }
+            
+            context("JSON API") {
+                it("should transform the keys") {
+                    let dog = Dog(id: "22", dogName: "Joan")
+                    let user = JUser(id: "11", userName: "Alex", doggyDog: dog)
+                    let serialized = SnakecaseTransformer(wrapped: JSONAPISerializer(user)).serialize() as! [String: AnyObject]
+                    let json = JSON(serialized)
+                    
+                    let data = json["data"].dictionaryValue
+                    
+                    do {
+                        let attributes = data["attributes"]!.dictionaryValue
+                        expect(attributes["user_name"]?.string).to(equal("Alex"))
+                    }
+
+                    do {
+                        let relationships = data["relationships"]?.dictionaryValue
+                        
+                        let dog = relationships?["doggy_dog"]?.dictionaryValue
+                        let dogData = dog?["data"]?.dictionaryValue
+                        expect(dogData?["id"]?.string).to(equal("22"))
+                        expect(dogData?["type"]?.string).to(equal(Dog.type))
+                    }
+                    
+                    do {
+                        let included = json["included"].arrayValue
+                        let dog = included.first?.dictionaryValue
+                        expect(dog?["id"]?.string).to(equal("22"))
+
+                        let attributes = dog?["attributes"]?.dictionaryValue
+                        expect(attributes?["dog_name"]?.string).to(equal("Joan"))
+                    }
+                }
+            }
+            
+            context("JSON API Links") {
+                it("should transform the keys") {
+                    let object = JSONAPILink.Object(href: "test", meta: friend)
+                    let serialized = UppercaseTransformer(wrapped: object).serialize() as! [String: AnyObject]
+                    expect(serialized["href"]).toNot(beNil())
+                    let meta = serialized["meta"] as? [String: AnyObject]
+                    expect(meta?["FRIENDS"]).toNot(beNil())
+                }
+            }
+        }
```

Comment:
We have transformer in transformer in: "should apply transformations in the right order".
IMO we shouldn't do it directly because doesn't really matter, that is a detail implementation of CustomSerializable for SerializationTransformers but people using kakapo shouldn't care about it because it's already done by kakapo with the protocol extension. Right?